go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -0,0 +1,1130 @@
+	for _, link := range links {
+		link := link
+		linkName := link.Attrs().Name
+		linkIdx := link.Attrs().Index
+		addresses, err := linkmanager.GetExternallyAvailableAddresses(link, c.v4, c.v6)
+		if err != nil {
+			return fmt.Errorf("unable to get link addresses for link %s: %v", linkName, err)
+		}
+		for _, address := range addresses {
+			if address.Label == linkmanager.GetAssignedAddressLabel(linkName) {
+				addressStr := address.String()
+				assignedAddr.Insert(addressStr)
+				assignedAddrStrToAddrs[addressStr] = address
+			}
+		}
+		filter, mask := filterRouteByLinkTable(linkIdx, getRouteTableID(linkIdx))
+		existingRoutes, err := util.GetNetLinkOps().RouteListFiltered(netlink.FAMILY_ALL, filter, mask)
+		if err != nil {
+			return fmt.Errorf("unable to get route list using filter (%s): %v", filter.String(), err)
+		}
+		for _, existingRoute := range existingRoutes {
+			route := routemanager.ConvertNetlinkRouteToRoute(existingRoute)
+			routeStr := route.String()
+			assignedIPRoutes.Insert(routeStr)
+			assignedIPRouteStrToRoutes[routeStr] = routemanager.RoutesPerLink{Link: link,
+				Routes: []routemanager.Route{route}}
+		}
+	}
+	filter, mask := filterRuleByPriority(rulePriority)
+	existingRules, err := util.GetNetLinkOps().RuleListFiltered(netlink.FAMILY_ALL, filter, mask)
+	if err != nil {
+		return fmt.Errorf("failed to list IP rules: %v", err)
+	}
+	for _, existingRule := range existingRules {
+		ruleStr := existingRule.String()
+		assignedIPRules.Insert(ruleStr)
+		assignedIPRulesStrToRules[ruleStr] = existingRule
+	}
+	if c.v4 {
+		ipTableV4Rules, err := c.iptablesManager.GetIPv4ChainRuleArgs(utiliptables.TableNAT, chainName)
+		if err != nil {
+			return fmt.Errorf("failed to list IPTable IPv4 rules: %v", err)
+		}
+		for _, rule := range ipTableV4Rules {
+			ruleStr := strings.Join(rule.Args, " ")
+			assignedIPTableV4Rules.Insert(ruleStr)
+			assignedIPTablesV4StrToRules[ruleStr] = rule
+		}
+	}
+	if c.v6 {
+		ipTableV6Rules, err := c.iptablesManager.GetIPv6ChainRuleArgs(utiliptables.TableNAT, chainName)
+		if err != nil {
+			return fmt.Errorf("failed to list IPTable IPv4 rules: %v", err)
+		}
+		for _, rule := range ipTableV6Rules {
+			ruleStr := strings.Join(rule.Args, " ")
+			assignedIPTableV6Rules.Insert(ruleStr)
+			assignedIPTablesV6StrToRules[ruleStr] = rule
+		}
+	}
+	expectedAddrs := sets.New[string]()
+	expectedIPRoutes := sets.New[string]()
+	expectedIPRules := sets.New[string]()
+	expectedIPTableV4Rules := sets.New[string]()
+	expectedIPTableV6Rules := sets.New[string]()
+	egressIPs, err := c.getAllEIPs()
+	if err != nil {
+		return err
+	}
+	node, err := c.nodeLister.Get(c.nodeName)
+	if err != nil {
+		return err
+	}
+	for _, egressIP := range egressIPs {
+		if len(egressIP.Status.Items) == 0 {
+			continue
+		}
+		for _, status := range egressIP.Status.Items {
+			if isValid := isEIPStatusItemValid(status, c.nodeName); !isValid {
+				continue
+			}
+			eIPNet, err := getFullMaskIPNet(status.EgressIP)
```

Comment:
This should probably go into util pkg, we have related methods there already `GetIPFullMaskString`/`GetIPFullMask` so it's probable we are doing something similar somewhere else.

Reply:
done